File: bulk/javascript/idor_01.js
// CWE-639: Insecure Direct Object Reference
const express = require('express');
const app = express();
app.get('/documents/:id', async (req, res) => {
  const doc = await getDocument(req.params.id); // no ownership check
  res.json(doc);
});
async function getDocument(id) { return { id }; }

File: bulk/java/Idor01.java
// CWE-639: Insecure Direct Object Reference
import org.springframework.web.bind.annotation.*;
@RestController
public class Idor01 {
    @GetMapping("/invoices/{id}")
    public Object getInvoice(@PathVariable Long id) {
        return fetchInvoice(id); // no ownership check
    }
    Object fetchInvoice(Long id) { return id; }
}
